tests/flow_validation/test_expression_eval_state.py

```diff
@@ -0,0 +1,133 @@
+import typing as t
+
+import pytest
+
+from globus_automate_client.models import FlowDefinition, FlowValidationError
+
+simple_eval_state = {
+    "StartAt": "ExpressionEval",
+    "States": {
+        "ExpressionEval": {
+            "Type": "ExpressionEval",
+            "Comment": "No info",
+            "Parameters": None,
+            "ResultPath": "$.some_path",
+            "End": True,
+        },
+    },
+}
+
+nested_parameters_with_json_path_syntax = {
+    "StartAt": "ExpressionEval",
+    "States": {
+        "ExpressionEval": {
+            "Type": "ExpressionEval",
+            "Comment": "No info",
+            "Parameters": {"nested_json_path.$": "$.some_path"},
+            "ResultPath": "$.some_path",
+            "End": True,
+        },
+    },
+}
+
+nested_parameters_with_invalid_json_path_key = {
+    "StartAt": "ExpressionEval",
+    "States": {
+        "ExpressionEval": {
+            "Type": "ExpressionEval",
+            "Comment": "No info",
+            "Parameters": {"invalid_json_path_indicator": "$.some_path"},
```

Comment:
I don't actually know for sure...is it possible to specify static parameters? For example, would this be acceptable?

```json
{"valid_key": "valid value"}
```

If that's acceptable, should we bar values that start with `"$."`?

```json
{"interest_calculated": "$.07"}
```

Reply:
I think you're right that we want to support static parameter values with JSONPath format. I can see someone trying to create/define a JSONPath value such as:

`{"computed_json_path": "$.dynamic_path_to_plugin_elsewhere"}`

I'll get rid of this restriction.